spec/services/calculators/housing_costs_calculator_spec.rb

```diff
@@ -282,6 +284,294 @@ module Calculators
+              expect(calculator.net_housing_costs).to eq((housing_cost_amount + monthly_cash_housing - housing_benefit_amount) / 2)
+            end
+          end
+
+          context 'board and lodging different values' do
+            let(:housing_cost_type) { 'board_and_lodging' }
+            let(:housing_cost_amount) { 1500.00 }
+            let(:housing_benefit_amount) { 100.00 }
+
+            it 'should record half the housing cost less the housing benefit' do
+              expect(calculator.gross_housing_costs).to eq 750.00 + monthly_cash_housing
+              expect(calculator.monthly_housing_benefit).to eq 100.00
+              expect(calculator.net_housing_costs).to eq((housing_cost_amount + monthly_cash_housing - housing_benefit_amount) / 2)
+            end
+          end
+
+          context 'rent' do
+            let(:housing_cost_type) { 'rent' }
+
+            it 'should record the full monthly housing costs' do
+              expect(calculator.gross_housing_costs).to eq 1200.00 + monthly_cash_housing
+              expect(calculator.monthly_housing_benefit).to eq 500.00
+              expect(calculator.net_housing_costs).to eq 700.00 + monthly_cash_housing
+            end
+
+            context 'when net cost is below housing cap' do
+              let(:housing_cost_amount) { 600.00 }
+              let(:housing_benefit_amount) { 200.00 }
+
+              it 'should return net as gross_cost minus housing_benefit' do
+                expect(calculator.gross_housing_costs).to eq 600.00 + monthly_cash_housing
+                expect(calculator.monthly_housing_benefit).to eq 200.0
+                expect(calculator.net_housing_costs).to eq 400.00 + monthly_cash_housing
+              end
+            end
+          end
+
+          context 'mortgage' do
+            let(:housing_cost_type) { 'mortgage' }
+
+            it 'should record the full housing costs less the housing benefit' do
+              expect(calculator.gross_housing_costs).to eq 1200.00 + monthly_cash_housing
+              expect(calculator.monthly_housing_benefit).to eq 500.00
+              expect(calculator.net_housing_costs).to eq 700.00 + monthly_cash_housing
+            end
+
+            context 'when net cost is below housing cap' do
+              let(:housing_cost_amount) { 600.00 }
+              let(:housing_benefit_amount) { 200.00 }
+
+              it 'should return net as gross_cost minus housing_benefit' do
+                expect(calculator.gross_housing_costs).to eq 600.00 + monthly_cash_housing
+                expect(calculator.monthly_housing_benefit).to eq 200.0
+                expect(calculator.net_housing_costs).to eq 400.00 + monthly_cash_housing
+              end
+            end
+          end
+        end
+      end
+    end
+
+    def monthly_cash_housing
+      @monthly_cash_housing ||= monthly_transaction_amount_by(operation: :debit, category: :rent_or_mortgage)
+    end
```

Comment:
is this a valid test if it is using the live codes `monthly_transaction_amount_by` method to come up with a figure that it plugs into the test, and then test if it is that figure?

Reply:
mmm, not sure. The test amounts created are a random figure between 10-100. What they are we don't care about. As they are randomly generated I thought this the easiest way to grab them. The tests above add the monthly cash amount to the monthly bank amounts, if they didn't exist or were `0` then the relevant caps wouldn't be hit above and the tests would fail. I've changed the hard coded bank amounts to accommodate for this. The bank amounts are in the outgoings table and easier to access than the cash transactions as they link up through two tables, so to avoid additional application code for just the sake of the test suite I thought it simpler to do it this way.